lib/do_volume/do_delete.js

```diff
@@ -80,36 +80,76 @@ function do_delete(subcmd, opts, args, cb) {
     }
 
     var context = {
-        volumeIds: args
+        volumeIds: args,
+        cli: this.top
     };
 
     vasync.pipeline({arg: context, funcs: [
-        function setup(ctx, next) {
-            common.cliSetupTritonApi({
-                cli: self.top
-            }, function onSetup(setupErr) {
-                next(setupErr);
-            });
+        common.cliSetupTritonApi,
+        function confirm(ctx, next) {
+            var promptMsg;
+
+            if (opts.yes) {
+                next();
+                return;
+            }
+
+            if (ctx.volumeIds.length === 1) {
+                promptMsg = format('Delete volume %s? [y/n] ',
+                    ctx.volumeIds[0]);
+            } else {
+                promptMsg = format('Delete %d volumes? [y/n] ',
+                    ctx.volumeIds.length);
+            }
+
+            common.promptYesNo({msg: promptMsg},
+                function onPromptAnswered(answer) {
+                    if (answer !== 'y') {
+                        console.error('Aborting');
+                        /*
+                         * Early abort signal.
+                         */
+                        next(true);
+                    } else {
+                        next();
+                    }
+                });
         },
         function deleteVolumes(ctx, next) {
             vasync.forEachParallel({
                 func: function doDeleteVolume(volumeId, done) {
-                    deleteVolume(volumeId, {
-                        wait: opts.wait,
+                    self.top.tritonapi.deleteVolume({
+                        id: volumeId,
+                        wait: opts.wait && opts.wait.length > 0,
+                        waitTimeout: opts.wait_timeout * 1000,
                         tritonapi: self.top.tritonapi
-                    }, done);
+                    }, function onVolDeleted(volDelErr) {
+                        if (volDelErr) {
```

Comment:
If deleting one of the volumes failed, I think we should collect those errors and ultimately exit non-zero. IOW, in `onDone` we should callback with a the volDelErr (if there was one), or a MultiErr if there were many.